packages/ui/src/components/organisms/TagList/TagList.tsx

```diff
@@ -52,8 +61,29 @@ const TagList = ({style, title, tags, defaultColor}: TagListProps) => {
     [tags],
   );
 
+  const renderEmptyState = useCallback(() => {
+    const _title =
+      (checkNullString(title) ? translator?.('Base_Data') : title) ?? 'data';
+    const lowerTitle = _title.toLowerCase();
+
+    const message =
+      translator != null
+        ? translator('Base_NoDataAvailable', {title: lowerTitle})
+        : `No ${lowerTitle} available.`;
+
+    return (
+      <Badge style={styles.badge} title={message} color={Colors.infoColor} />
+    );
+  }, [Colors.infoColor, title, translator]);
+
   if (visibleSortTags.length === 0) {
-    return null;
+    if (hideIfNull) {
+      return null;
+    } else {
+      return (
+        <View style={[styles.container, style]}>{renderEmptyState()}</View>
```

Comment:
It should also display the title if provided 

Maybe try to simplify with render below. Use useMemo to how if the component is in empty state
If it's empty and hideIfNull is true, return null
Else go to last return and modify render to render tags if component is not empty, render the text message if it's empty.